rct229/rulesets/ashrae9012019/section19/section19rule13.py

```diff
@@ -0,0 +1,255 @@
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+
+        zone_info = {}
+        for hvac_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b:
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                zone_info[zone_id_b] = {}
+                zone_info[zone_id_b][
+                    "are_all_hvac_sys_fan_objs_autosized"
+                ] = are_all_hvac_sys_fan_objs_autosized(rmi_b, hvac_id_b)
+                zone_info[zone_id_b]["proposed_supply_airflow"] = ZERO.FLOW
+                zone_info[zone_id_b][
+                    "all_design_setpoints_delta_Ts_are_per_reqs"
+                ] = ZERO.TEMPERATURE
+
+                # get `design_thermostat_cooling_setpoint` and `design_thermostat_heating_setpoint`
+                for zone_b in find_all(
+                    '$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")]',
+                    rmi_b,
+                ):
+                    zone_info[zone_id_b]["design_thermostat_cooling_setpoint"] = min(
+                        find_all(
+                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")].design_thermostat_cooling_setpoint',
+                            rmi_b,
+                        )
+                    )
+                    zone_info[zone_id_b]["design_thermostat_heating_setpoint"] = max(
+                        find_all(
+                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")].design_thermostat_heating_setpoint',
+                            rmi_b,
+                        )
+                    )
+
+                    for space_b in getattr_(zone_b, "Zone", "spaces"):
+                        zone_has_lab_space = (
+                            True
+                            if (
+                                getattr_(space_b, "Space", "lighting_space_type")
+                                == LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                            )
+                            else False
+                        )
+
+                    # calculate proposed_supply_airflow
+                    for terminal_p in find_all(
+                        '$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")].terminals[*]',
+                        rmi_p,
+                    ):
+                        zone_info[zone_id_b]["proposed_supply_airflow"] += getattr_(
+                            terminal_p, "Terminal", "primary_airflow"
+                        )
+
+                for terminal_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
```

Comment:
I prefer to have below logics happen in the `calc_val` function so it can be part of reporting values.

Reply:
I believe this can't be done in the ```HVACRule``` level. Please let me know if I am missing something.

Reply:
However, I found that ```terminal_b``` should be ```terminal_id_b``` and made some updates this part.

Reply:
Addressed!